docs/services.rst

```diff
@@ -0,0 +1,307 @@
+    s.run()  # blocks forever
+
+
+.. note::
+    If you'd like to try out *switchy* routes alongside your existing
+    XML dialplan (assuming you've added the :ref:`park only <parkonly>`
+    context in your existing config) you can either pass in
+    ``{"Caller-Context": "switchy"}`` as a ``guard`` or you can load
+    the router with:
+
+    ``s.apps.load_app(router, app_id='switchy', header='Caller-Context')``
+
+
+Replicating XML dialplan features
+*********************************
+The main difference with using *switchy* for call control is that
+everything is processed at **runtime** as opposed to having separate *parse*
+and *execute* phases.
+
+Retrieving Variables
+^^^^^^^^^^^^^^^^^^^^
+`Accessing variable`_ values from *FreeSWITCH* is already built into
+*switchy*'s :doc:`sessionapi` using traditional `getitem`_ access.
+
+Basic Logic
+^^^^^^^^^^^
+As a first note, you can accomplish any "logical" *field* pattern match
+either directly in Python or by the *regex* expression to ``Router.route``:
+
+Here is the equivalent of the logical `AND`_ example:
+
+.. code-block:: python
+
+    from datetime import datetime
+
+    @router.route('^500$')
+    def on_sunday(sess, match, router, profile='internal', did='500'):
+        """On Sunday no one works in support...
+        """
+        did = '531' if datetime.today().weekday() == 6 else did
+        sess.bridge('{}@example.com'.format(did), profile=profile)
+
+And the same for logical `OR`_ example:
+
+.. code-block:: python
+
+    import re
+
+    # by regex
+    @router.route('^500$|^502$')
+    def either_ext(sess, match, router):
+        sess.answer()
+        sess.playback('ivr/ivr-welcome_to_freeswitch.wav')
+
+    # by if statement
+    @router.route('^.*$')
+    def match(sess, match, router):
+        if re.match("^Michael\s*S?\s*Collins", sess['variable_caller_id_name']) or\
+                re.match("^1001|3757|2816$", sess['variable_caller_id_number']):
+            sess.playback("ivr/ivr-dude_you_rock.wav")
+        else:
+            sess.playback("ivr/ivr-dude_you_suck.wav")
+
+
+Nesting logic
+^^^^^^^^^^^^^
+`Nested conditions`_ Can be easily accomplished using plain old `if statements`_:
+
+.. code-block:: python
+
+    @router.route('^1.*(\d)$')
+    def play_wavfile(sess, match, router):
+        # get the last digit
+        last_digit = match.groups()[0]
+
+        # only play the extra file when last digit is '3'
+        if last_digit == '3':
+            sess.playback('foo.wav')
+
+        # always played if the first digit is '1'
+        sess.playback('bar.wav')
+
+
+Break on true
+^^^^^^^^^^^^^
+Halting route execution (known as `break on true`_) can be done currently by returning
+``True`` from your routing function:
+
+.. code-block:: python
+
+    @router.route('^1.*(\d)$')
+    def play_wavfile(sess, match, router):
+        if not sess['Caller-Destination-Number'] == "1100":
+            return True  # stop all further routing
```

Comment:
I see how 'break on true' and returning true seem to go well, but also wondered if a more pythonic way of doing this is to raise an StopRouting exception.